src/plugins/vis_type_timeseries/public/application/components/lib/convert_series_to_vars.js

```diff
@@ -54,6 +54,18 @@ export const convertSeriesToVars = (series, model, dateFormat = 'lll', getConfig
         };
         set(variables, varName, data);
         set(variables, `${_.snakeCase(row.label)}.label`, row.label);
+
+        if (row.labelFormatted) {
+          let val;
+
+          if (row.labelFormatted === 'true' || row.labelFormatted === 'false') {
+            val = row.labelFormatted;
+          } else {
+            val = moment(row.label).format(dateFormat);
+          }
```

Comment:
what if we just check for if the `labelFormatted` is actually a date?
```
if(moment.isDate(row.labelFormatted)) {
val =  moment(row.labelFormatted).format(dateFormat);
}else {
val = row.labelFormatted
}
```

Reply:
In such a case we'll be able to format only moment known date formats, and the case you described above will not work.. (will render an elastic stored value).
Are you OK with that?

Reply:
yes because I'm afraid that if not going through this way we can easily format numbers that are not dates if somehow a number field has also a key_as_string

Reply:
ok, agree